ee/tabby-webserver/src/service/github_repository_provider.rs

```diff
@@ -142,11 +161,20 @@ impl GithubRepositoryProviderService for GithubRepositoryProviderServiceImpl {
             .filter_map(|provider| Some((provider.id.to_string(), provider.access_token?)))
             .collect();
 
-        let urls = self
+        let mut repos = self
             .list_github_provided_repositories_by_provider(vec![], None, None, None, None)
-            .await?
+            .await?;
+
+        // Deduplicate by vendor ID
+        let mut vendor_ids = HashSet::new();
+        repos.retain(|repo| vendor_ids.insert(repo.vendor_id.clone()));
```

Comment:
nit: extract a function: `deduplicate_github_repositories()` for better readability